### PR TITLE
Upgrading SC instance through the SCMU

### PR DESCRIPTION
**Pull request by HEskandari**

### CODE REVIEW

File: src/ServiceControl.Config/Commands/UpgradeServiceControlInstanceCommand.cs [hunk 1]
```diff
@@ -171,15 +181,103 @@ public override async Task ExecuteAsync(InstanceDetailsViewModel model)
                 }
             }
 
+            if (!instance.VersionHasServiceControlAuditFeatures)
+            {
+                auditViewModel = auditUpgradeViewModelFactory(instance.Name);
+                if (windowManager.ShowInnerDialog(auditViewModel) == true)
+                {
+                    upgradeOptions.InstallNewAuditSidecar = true;
```
Comment (mikeminutillo) [inline src/ServiceControl.Config/Commands/UpgradeServiceControlInstanceCommand.cs]: Does this need to be added to the upgrade options? It only ever gets referred to from this method

File: src/ServiceControl.Config/Commands/UpgradeServiceControlInstanceCommand.cs [hunk 2]
```diff
@@ -171,15 +181,103 @@ public override async Task ExecuteAsync(InstanceDetailsViewModel model)
                 }
             }
 
+            if (!instance.VersionHasServiceControlAuditFeatures)
```
Comment (mikeminutillo) [inline src/ServiceControl.Config/Commands/UpgradeServiceControlInstanceCommand.cs]: What about the case where audit was already disabled?

File: src/ServiceControl.Config/UI/Upgrades/AddNewAuditInstanceViewModelValidator.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                .WithMessage(Validations.MSG_MUST_BE_UNIQUE, "Paths")
+                .When(x => x.SubmitAttempted);
+
+            RuleFor(x => x.ServiceControlAudit.AuditForwarding)
+                .NotNull().WithMessage(Validations.MSG_SELECTAUDITFORWARDING);
+
+            RuleFor(x => x.ServiceControlAudit.AuditQueueName)
+                .NotEmpty()
+                .NotEqual(x => x.ServiceControlAudit.AuditForwardingQueueName).WithMessage(Validations.MSG_UNIQUEQUEUENAME, "Audit Forwarding")
+                .When(x => x.SubmitAttempted && x.ServiceControlAudit.AuditQueueName != "!disable");
```
Comment (mikeminutillo) [inline src/ServiceControl.Config/UI/Upgrades/AddNewAuditInstanceViewModelValidator.cs]: We wouldn't have an Audit Queue `!disable` in an Audit Instance. You'd just remove the audit instance

File: src/ServiceControl.Config/Commands/UpgradeServiceControlInstanceCommand.cs (hunk 1 above)
Comment (HEskandari) [reply src/ServiceControl.Config/Commands/UpgradeServiceControlInstanceCommand.cs]: Possibly not. I wasn't sure if it is needed for other purposes. I can remove for now.

File: src/ServiceControl.Config/Commands/UpgradeServiceControlInstanceCommand.cs (hunk 2 above)
Comment (HEskandari) [reply src/ServiceControl.Config/Commands/UpgradeServiceControlInstanceCommand.cs]: That's in the PoA to be done ;) We can include the check it in this PR though.

File: src/ServiceControl.Config/UI/Upgrades/AddNewAuditInstanceViewModelValidator.cs (hunk 3 above)
Comment (HEskandari) [reply src/ServiceControl.Config/UI/Upgrades/AddNewAuditInstanceViewModelValidator.cs]: I've copied the validator code verbatim from the existing validation logic of the Add screen. Should it be removed altogether or just here?

**Pull request closed without merging**